sw/device/lib/dif/dif_sysrst_ctrl_unittest.cc

```diff
@@ -240,5 +240,391 @@ TEST_F(UlpWakeupConfigTest, Success) {
   EXPECT_DIF_OK(dif_sysrst_ctrl_ulp_wakeup_configure(&sysrst_ctrl_, config_));
 }
 
+class UlpWakeupSetEnabledTest : public SysrstCtrlTest {};
+
+TEST_F(UlpWakeupSetEnabledTest, NullArgs) {
+  EXPECT_DIF_BADARG(
+      dif_sysrst_ctrl_ulp_wakeup_set_enabled(nullptr, kDifToggleEnabled));
+}
+
+TEST_F(UlpWakeupSetEnabledTest, BadEnabled) {
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_ulp_wakeup_set_enabled(
+      &sysrst_ctrl_, static_cast<dif_toggle_t>(2)));
+}
+
+TEST_F(UlpWakeupSetEnabledTest, Success) {
+  EXPECT_WRITE32(SYSRST_CTRL_ULP_CTL_REG_OFFSET, 0);
+  EXPECT_DIF_OK(dif_sysrst_ctrl_ulp_wakeup_set_enabled(&sysrst_ctrl_,
+                                                       kDifToggleDisabled));
+
+  EXPECT_WRITE32(SYSRST_CTRL_ULP_CTL_REG_OFFSET, 1);
+  EXPECT_DIF_OK(
+      dif_sysrst_ctrl_ulp_wakeup_set_enabled(&sysrst_ctrl_, kDifToggleEnabled));
+}
+
+class UlpWakeupGetEnabledTest : public SysrstCtrlTest {};
+
+TEST_F(UlpWakeupGetEnabledTest, NullArgs) {
+  dif_toggle_t is_enabled;
+  EXPECT_DIF_BADARG(
+      dif_sysrst_ctrl_ulp_wakeup_get_enabled(nullptr, &is_enabled));
+  EXPECT_DIF_BADARG(
+      dif_sysrst_ctrl_ulp_wakeup_get_enabled(&sysrst_ctrl_, nullptr));
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_ulp_wakeup_get_enabled(nullptr, nullptr));
+}
+
+TEST_F(UlpWakeupGetEnabledTest, Success) {
+  dif_toggle_t is_enabled;
+
+  EXPECT_READ32(SYSRST_CTRL_ULP_CTL_REG_OFFSET, 0);
+  EXPECT_DIF_OK(
+      dif_sysrst_ctrl_ulp_wakeup_get_enabled(&sysrst_ctrl_, &is_enabled));
+  EXPECT_EQ(is_enabled, kDifToggleDisabled);
+
+  EXPECT_READ32(SYSRST_CTRL_ULP_CTL_REG_OFFSET, 1);
+  EXPECT_DIF_OK(
+      dif_sysrst_ctrl_ulp_wakeup_get_enabled(&sysrst_ctrl_, &is_enabled));
+  EXPECT_EQ(is_enabled, kDifToggleEnabled);
+}
+
+class PinsSetInvertedTest : public SysrstCtrlTest {
+ protected:
+  uint32_t _pins = kDifSysrstCtrlPinKey1Out | kDifSysrstCtrlPinPowerButtonIn;
```

Comment:
You have the `_` in the wrong place. Though I would have just made this a namespace-scope constexpr variable...

Reply:
Oops. Fixed.